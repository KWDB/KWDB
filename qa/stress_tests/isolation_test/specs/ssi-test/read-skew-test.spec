# test read skew

setup
{
  CREATE TABLE IF NOT EXISTS foo (
	key		int PRIMARY KEY,
	value	int
  );

  INSERT INTO foo VALUES (50, 50);
}

teardown
{
  DROP TABLE foo;
}

session s1
step s1		{ BEGIN transaction isolation level SERIALIZABLE; }
step s1r	{ SELECT * FROM foo; }
step s1w1	{ UPDATE foo SET key=25; }
step s1w2	{ UPDATE foo SET value=75; }
step s1c	{ COMMIT; }

session s2
step s2		{ BEGIN transaction isolation level SERIALIZABLE; }
step s2r	{ SELECT * FROM foo; }
step s2c	{ COMMIT; }

permutation s1 s2 s1r s2r s1w1 s1w2 s1c s2r s2c 

