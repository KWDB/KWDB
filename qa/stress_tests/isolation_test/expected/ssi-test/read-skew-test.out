Parsed test spec with 2 sessions

starting permutation: s1 s2 s1r s2r s1w1 s1w2 s1c s2r s2c
step s1: BEGIN transaction isolation level SERIALIZABLE;  finished.
step s2: BEGIN transaction isolation level SERIALIZABLE;  finished.
step s1r: SELECT * FROM foo;  finished.
key|value
---+-----
 50|   50
(1 row)

step s2r: SELECT * FROM foo;  finished.
key|value
---+-----
 50|   50
(1 row)

step s1w1: UPDATE foo SET key=25;  finished.
step s1w2: UPDATE foo SET value=75;  finished.
step s1c: COMMIT;  finished.
step s2r: SELECT * FROM foo;  finished.
key|value
---+-----
 50|   50
(1 row)

step s2c: COMMIT;  finished.
